assets/js/TableView.test.js

```diff
@@ -24,20 +26,35 @@ describe('TableView', () => {
     return jsonResponseMock(data);
   }
 
-  it('Fetches, renders, and edits cells', (done) => {
+
+  // This test has been re-written and watered-down as wrapping the whole thing in
+  // DragDropContextProvider makes it nigh impossible to do state testing
+  it('Fetches, renders, edits cells, sorts columns and reorders columns', (done) => {
 
     var api = {
       render: makeRenderResponse(0, 2, 1000)
     };
 
-    const tree = mount(<TableView id={100} revision={1} api={api}/>)
+    // Mocks table-related operations for testing
+    let addCellEditMock = jest.fn();
+    mockAddCellEdit(addCellEditMock);
+    let updateSortMock = jest.fn();
+    mockSortColumn(updateSortMock);
+    let reorderColumnsMock = jest.fn();
+    mockReorderColumns(reorderColumnsMock);
+
+
+    const tree = mount(
+        <DragDropContextProvider backend={TestBackend}>
```

Comment:
We've recently upgraded to Enzyme 3. Now `shallow()` will call lifecycle method so we shouldn't need `mount()`. Which means the `DataGrid` child will not be mounted, so we shouldn't need `DragDropContextProvider`